### PR TITLE
Demo STF tests added. Minor improvements to STF.

### PR DESCRIPTION
**Pull request by i-Alex**

### CODE REVIEW

File: qa/demo.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        )
+
+        creation_amount = 100  # Zen
+        withdrawal_epoch_length = 10
+        sc_creation_info = SCCreationInfo(mc_node, creation_amount, withdrawal_epoch_length)
+        accounts = generate_secrets("seed", 1)
+        vrf_keys = generate_vrf_secrets("seed", 1)
+        genesis_account = accounts[0]
+        vrf_key = vrf_keys[0]
+        withdrawal_certificate_data = generate_withdrawal_certificate_data("seed", 7, 5)
```
Comment (albertog78) [inline qa/demo.py:65]: rename to certificate_proof_info

File: qa/demo.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        sc_block_ids = generate_next_blocks(sc_node, "first node", 4)
+        print("\nGenerating Withdrawal Certificate...\n")
+
+        attempts = 20
+        while mc_node.getmempoolinfo()["size"] == 0 and attempts > 0:
+            if attempts % 4 == 0:
+                print("Wait for withdrawal certificate in MC memory pool...")
+            time.sleep(10)
+            attempts -= 1
+            sc_node.block_best()  # just a ping to SC node. For some reason, STF can't request SC node API after a while idle.
```
Comment (albertog78) [inline qa/demo.py:236]: remove

File: qa/demo.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        print "MC Block id - {}\n".format(mc_block_4)
+
+        self.pause()
+
+        # Get SC balances changes
+        sc_balance = sc_node.wallet_balance()["result"]
+        print("\nSC wallet balance in satoshi: {}".format(
+            json.dumps(sc_balance, indent=4, sort_keys=True)))
+        boxes_balances = sc_node.wallet_allBoxes()["result"]
+        print("\nSC wallet boxes: {}".format(json.dumps(boxes_balances, indent=4, sort_keys=True)))
```
Comment (albertog78) [inline qa/demo.py:260]: verify mc address balance is considering BT

Comment (i-Alex) [reply qa/demo.py:269]: done

**Review by albertog78 — APPROVED** [review]

**Pull request merged**